qt/widgets/common/inc/MantidQtWidgets/Common/QtPropertyBrowser/qteditorfactory.h

```diff
@@ -778,16 +867,98 @@ class QtSliderFactoryPrivate : public EditorFactoryPrivate<QSlider> {
   void slotSetValue(int value);
 };
 
-class QtSpinBoxFactoryPrivate : public EditorFactoryPrivate<QSpinBox> {
-  QtSpinBoxFactory *q_ptr;
-  Q_DECLARE_PUBLIC(QtSpinBoxFactory)
+template <class SpinBox>
+class QtSpinBoxFactoryPrivateBase : public EditorFactoryPrivate<SpinBox> {
 public:
+  // This q_ptr is public due to the base class QtSpinBoxFactoryBase needing
+  // access to it.
+  QtSpinBoxFactoryBase<SpinBox> *q_ptr;
```

Comment:
I don't quite understand the comment here. Is this not the base class we are defining here? And if so can this be protected rather than private?

Reply:
Actually I have now seen where this needs to be accessed further down. What has changed to mean this needs to be private now though.

Reply:
So the Q_DELCARE_PUBLIC / Q_DECLARE_PRIVATE Macros usually handle this but isn't implemented for templated classes. So in order for the constructor of the base class to see this q_ptr we had to make it public... It was all a bit confusing